cudax/include/cuda/experimental/__execution/fwd.cuh

```diff
@@ -26,13 +26,24 @@
 #include <cuda/std/__type_traits/type_list.h>
 
 #include <cuda/experimental/__detail/utility.cuh>
-#include <cuda/experimental/__execution/meta.cuh>
+#include <cuda/experimental/__execution/type_traits.cuh>
 #include <cuda/experimental/__execution/visit.cuh>
 
 #include <cuda/experimental/__execution/prologue.cuh>
 
-namespace cuda::experimental::execution
+namespace cuda::experimental
 {
+// so we can refer to the cuda::experimental::__detail namespace below
+namespace __detail
+{
+}
+namespace execution
+{
+namespace __detail
+{
+using namespace cuda::experimental::__detail; // // NOLINT(misc-unused-using-decls)
```

Comment:
Why are we using the namespace inside itself?

Reply:
We have `cuda::experimental::__detail` and `cuda::experimental::execution::__detail`. This brings the former into the latter. Otherwise I have to always specify which `__detail` namespace I mean.